src/core/table/table.jsx

```diff
@@ -0,0 +1,268 @@
+import React, { useState, useMemo } from 'react';
+import PropTypes from 'prop-types'
+import _ from 'lodash';
+
+import Box from '@mui/material/Box';
+import {
+    Table as MuiTable,
+    TableBody,
+    TableCell,
+    TableContainer,
+    TableHead,
+    TablePagination,
+    TableRow,
+    TableSortLabel,
+    Paper,
+    Checkbox,
+    Radio,
+    CircularProgress
+} from '@mui/material';
+import { visuallyHidden } from '@mui/utils';
+
+/*
+    @columns: Array of objects
+    @column: {
+        id: Unique identifier
+        accessor: Property to be accessed
+        label: Header label
+        template: custom template
+    }
+*/
+
+export const Table = ({
+    data,
+    columns = [],
+    loading = false,
+    onRowClick = _.noop,
+    onRowSelect = _.noop,
+    defaultOrder = 'asc',
+    defaultOrderBy = '',
+    multiSelect = false
+}) => {
+    const [order, setOrder] = useState(defaultOrder);
+    const [orderBy, setOrderBy] = useState(defaultOrderBy);
+    const [selected, setSelected] = React.useState([]);
```

Comment:
useState is already imported

```suggestion
    const [selected, setSelected] = useState([]);
```